glib/src/commonMain/kotlin/io/github/cgr/glib/LazyList.kt

```diff
@@ -0,0 +1,162 @@
+package io.github.cgr.glib
+
+class LazyList(private val generator: Iterator<Any?>) : List<Any?> {
+
+    constructor(generator: () -> Any?) : this(object : Iterator<Any?> {
```

Comment:
I feel like rather than a constructor, this should be a method on the companion with a descriptive name. Something like `LazyList.repeat`